builds/misc/templates/build-images.yaml

```diff
@@ -0,0 +1,456 @@
+parameters:
+  - name: 'E2EBuild'
+    type: boolean
+    default: false
+
+stages:
+
+  ################################################################################
```

Comment:
I added this additional stage to prevent building Images if no source code changes

Reply:
If the code changes are under `test/Microsoft.Azure.Devices.Edge.Test*` what will happen? Where will the end-to-end test checkin pipeline get images?